src/engine/srv.c

```diff
@@ -1261,18 +1262,29 @@ dss_sys_db_init()
 		return rc;
 	}
 
-	if (lmm_db_path == NULL) {
-		D_ERROR("DAOS_LMM_DB_PATH need be configured\n");
+	if (dss_nvme_conf == NULL) {
+		D_ERROR("nvme conf path not set\n");
 		return -DER_INVAL;
 	}
 
+	D_STRNDUP(nvme_conf_path, dss_nvme_conf, PATH_MAX);
```

Comment:
Why do we need to duplicate the dss_nvme_conf here?

Reply:
from manpage: "Both dirname() and basename() may modify the contents of path, so it may be desirable to pass a copy when calling one of these functions." , dss_nvme_conf is const and dirname takes char* (we can cast the const out but seems like the better behaviour is to copy so you don't have any chance of updating the original).

Reply:
have removed it anyway and just used a cast

Reply:
Done